src/runtime/policy/policy.ts

```diff
@@ -279,6 +290,65 @@ export class PolicyField {
       label: label === 'raw' ? '' : label,
     };
   }
+
+  getRestrictedFields(parentType: Type) {
+    const field = parentType.getEntitySchema().fields[this.name];
+    return [this.name, this.restrictField(field)];
+  }
+
+  private restrictField(field) {
+    switch (field.kind) {
+      case 'kotlin-primitive':
+      case 'schema-primitive': {
+        assert(this.subfields.length === 0);
+        return field;
+      }
+      case 'schema-collection':
+      case 'schema-ordered-list':
+      case 'schema-nested': {
+        return FieldType.create(
+          {kind: field.kind, schema: this.restrictField(field.schema)}
+        );
+      }
+      case 'schema-reference': {
+        return FieldType.create({
+          kind: 'schema-reference', schema: {
+            ...field.schema,
+            model: this.restrictedEntityType(field.getEntityType())
+          }
+        });
+      }
+      case 'type-name':
+      case 'schema-inline': {
+        let entityType = field.getEntityType();
```

Comment:
if you know the type of the field you want to create it's better to directly use the explicit ctor, e.g.:
new ReferenceType(this.restrictedEntityType(entityType))

Reply:
The constructors are in `internal/schema-fields.ts`. Are they still considered accessible?

Reply:
you'd need to add `ReferenceField` to the current export statement in lib-types.ts:
`export {FieldType, PrimitiveField, OrderedListField} from './internal/schema-field.js';`
(and same for the inline fields as well)